.github/workflows/ci.yml

```
@@ -1,346 +1,208 @@
+          replace_once(session, '\tasync setHeadedMode(headed: boolean): Promise<void> {', '\tasync setHeadedMode(headed: boolean, beforeNavigation?: PreNavigationPageHook): Promise<void> {')
+          replace_once(session, '\t\tawait this.attachSecurityHooks(newPage);\n\n\t\t// Re-install auto-dialog handler on new page', '\t\tawait this.attachSecurityHooks(newPage);\n\t\tif (beforeNavigation) await beforeNavigation(newPage);\n\n\t\t// Re-install auto-dialog handler on new page')
+          replace_once(session, '\tasync openPage(url: string): Promise<TaloxPageState> {', '\tasync openPage(url: string, beforeNavigation?: PreNavigationPageHook): Promise<TaloxPageState> {')
+          replace_once(session, '\t\tawait this.attachSecurityHooks(page);\n\n\t\t// Install auto-dialog handler on new page', '\t\tawait this.attachSecurityHooks(page);\n\t\tif (beforeNavigation) await beforeNavigation(page);\n\n\t\t// Install auto-dialog handler on new page')
+
+          controller = "src/core/controller/TaloxController.ts"
+          replace_once(controller, '''\tprivate async setupOriginHeaders(page: import("playwright-core").Page): Promise<void> {\n\t\tif (!this.originHeaderConfig) return;\n\t\tconst headers = new OriginHeaders(this.originHeaderConfig);\n\t\tthis.originHeaders = headers;\n\t\tawait headers.install(page);\n\t}''', '''\tprivate async setupOriginHeaders(page: import("playwright-core").Page): Promise<void> {\n\t\tif (!this.originHeaderConfig) return;\n\t\tthis.originHeaders ??= new OriginHeaders(this.originHeaderConfig);\n\t\tawait this.originHeaders.installSessionPage(page);\n\t}''')
+          replace_once(controller, '\t\tawait this._session.setHeadedMode(headed);', '\t\tawait this._session.setHeadedMode(headed, (page) => this.setupOriginHeaders(page));')
+          replace_once(controller, '\t\tconst state = await this._session.openPage(url);', '\t\tconst state = await this._session.openPage(url, (page) => this.setupOriginHeaders(page));')
+
+          Path("tests/unit/OriginHeadersSessionLifecycle.test.ts").write_text(r'''import { describe, expect, it, vi } from "vitest";
+import { OriginHeaders } from "../../src/core/OriginHeaders.js";
```

actionlint (1.7.12)

[error] could not parse as YAML: could not find expected ':'

(syntax-check)